nifi-nar-bundles/nifi-ccda-bundle/nifi-ccda-processors/src/main/java/org/apache/nifi/processors/ccda/ExtractCCDAAttributes.java

```diff
@@ -0,0 +1,329 @@
+        final StopWatch stopWatch = new StopWatch(true);
+
+        // Load required MDHT packages
+        System.setProperty( "org.eclipse.emf.ecore.EPackage.Registry.INSTANCE",
+                "org.eclipse.emf.ecore.impl.EPackageRegistryImpl" );
+        CDAPackage.eINSTANCE.eClass();
+        HITSPPackage.eINSTANCE.eClass();
+        CCDPackage.eINSTANCE.eClass();
+        ConsolPackage.eINSTANCE.eClass();
+        IHEPackage.eINSTANCE.eClass();
+        stopWatch.stop();
+        getLogger().debug("Loaded packages in {}", new Object[] {stopWatch.getDuration(TimeUnit.MILLISECONDS)});
+
+        // Initialize JEXL
+        jexl = new JexlBuilder().cache(1024).debug(false).silent(true).strict(false).create();
+        jexlCtx = new MapContext();
+
+        getLogger().debug("Loading mappings");
+        loadMappings(); // Load CDA mappings for parser
+
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+        Map<String, String> attributes = new TreeMap<String, String>(); // stores CDA attributes
+        getLogger().info("Processing CCDA");
+
+        FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        if(processMap.isEmpty()) {
+            getLogger().error("Process Mapping is not loaded");
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final Boolean skipValidation = context.getProperty(SKIP_VALIDATION).asBoolean();
+
+        final StopWatch stopWatch = new StopWatch(true);
+
+        ClinicalDocument cd = loadDocument(session.read(flowFile), skipValidation); // Load and optionally validate CDA document
+
+        getLogger().debug("Loaded document for {} in {}", new Object[] {flowFile, stopWatch.getElapsed(TimeUnit.MILLISECONDS)});
+
+        getLogger().debug("Processing elements");
+        processElement(null, cd, attributes); // Process CDA element using mapping data
+
+        flowFile = session.putAllAttributes(flowFile, attributes);
+
+        stopWatch.stop();
+        getLogger().info("Successfully processed {} in {}", new Object[] {flowFile, stopWatch.getDuration(TimeUnit.MILLISECONDS)});
+        if(getLogger().isDebugEnabled()){
+            for (Entry<String, String> entry : attributes.entrySet()) {
+                getLogger().debug("Attribute: {}={}", new Object[] {entry.getKey(), entry.getValue()});
+            }
+        }
+
+        session.transfer(flowFile, REL_SUCCESS);
+
+    }
+
+    /**
+     * Process elements children based on the parser mapping.
+     * Any String values are added to attributes
+     * For List, the processList method is called to iterate and process
+     * For an Object this method is called recursively
+     * While adding to the attributes the key is prefixed by parent
+     * @param parent    Parent key for this element, used as a prefix for attribute key
+     * @param element   Element to be processed

```

Comment:
After the most recent changes looks like the javadoc is out of date because the attributes map isn't mentioned.